crates/astria-sequencer/src/app/mod.rs

```diff
@@ -747,13 +747,15 @@ impl App {
             .get_chain_id()
             .await
             .context("failed to get chain ID from state")?;
+        let sudo_address = self
```

Comment:
you can move this to the `end_block` call so you don't have to pass it as a parameter. `state_tx` implements StateRead

Reply:
I figured it would be better if it failed earlier than later (shouldn't fail), and this provides the ability to add fork choice logic elsewhere. So switching back to proposer is a couple of lines of code here without modifying the end_block function signature.